pkg/controllers/scheduler/scheduler.go

```diff
@@ -361,6 +361,38 @@ func (s *Scheduler) prepareToSchedule(
 ) {
 	logger := klog.FromContext(ctx)
 
+	// check migration config
+	// We must check migration configuration before the judgement of whether to skip scheduling.
+	// Because whether AppliedMigrationConfigurationAnnotation changes is one of the factors
+	// that affects whether rescheduling is needed.
```

Comment:
```suggestion
	// update applied migration config
	// We must update migration configuration before the checking scheduling triggers because
	// the former is part of the latter.
```